packages/x-tree-view/src/TreeItem/TreeItem.tsx

```diff
@@ -187,7 +187,16 @@ export const TreeItem = React.forwardRef(function TreeItem(
     icon: inSlots?.icon,
   };
 
-  const expandable = Boolean(Array.isArray(children) ? children.length : children);
+  const isExpandable = (reactChildren: React.ReactNode) => {
+    if (Array.isArray(reactChildren)) {
+      return (
+        reactChildren.length > 0 &&
+        reactChildren.some((child) => (Array.isArray(child) ? child.length > 0 : true))
```

Comment:
```suggestion
        reactChildren.some(isExpandable)
```

I think this change would allow us to support weird scenarios like

```tsx
<TreeItem>
  {[[]]}
</TreeItem>
```

Reply:
Very clever, committed.